src/.editorconfig

```diff
@@ -0,0 +1,271 @@
+csharp_space_between_parentheses = false
+csharp_space_before_colon_in_inheritance_clause = true
+csharp_space_after_colon_in_inheritance_clause = true
+csharp_space_around_binary_operators = before_and_after
+csharp_space_between_method_declaration_parameter_list_parentheses = false
+csharp_space_between_method_declaration_empty_parameter_list_parentheses = false
+csharp_space_between_method_declaration_name_and_open_parenthesis = false
+csharp_space_between_method_call_parameter_list_parentheses = false
+csharp_space_between_method_call_empty_parameter_list_parentheses = false
+csharp_space_between_method_call_name_and_opening_parenthesis = false
+csharp_space_after_comma = true
+csharp_space_before_comma = false
+csharp_space_after_dot = false
+csharp_space_before_dot = false
+csharp_space_after_semicolon_in_for_statement = true
+csharp_space_before_semicolon_in_for_statement = false
+csharp_space_around_declaration_statements = false
+csharp_space_before_open_square_brackets = false
+csharp_space_between_empty_square_brackets = false
+csharp_space_between_square_brackets = false
+
+# Wrapping preferences
+csharp_preserve_single_line_statements = true
+csharp_preserve_single_line_blocks = true
+
+################################################################
+# .NET / C# Code Style preferences                             #
+################################################################
+
+# 'using' directive preferences
+## IDE0065 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0065)
+csharp_using_directive_placement = outside_namespace:warning
+
+# Code-block preferences
+## IDE0063 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0063)
+csharp_prefer_simple_using_statement = true:suggestion
+## IDE0011 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0011)
+csharp_prefer_braces = true:suggestion
+## IDE0160 & IDE0161 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161)
+csharp_style_namespace_declarations = file_scoped:warning
+## IDE0200 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0200)
+csharp_style_prefer_method_group_conversion = true:warning
+## IDE0210 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0210)
+csharp_style_prefer_top_level_statements = true:suggestion
+## IDE0290 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0290)
+csharp_style_prefer_primary_constructors = true:suggestion
+
+# Expression-bodied members
+## IDE0021 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0021)
+csharp_style_expression_bodied_constructors = when_on_single_line:suggestion
+## IDE0022 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0022)
+csharp_style_expression_bodied_methods = when_on_single_line:suggestion
```

Comment:
I totally agree, which is why I've only given it the `suggestion` priority. It won't cause any warnings or complaints about methods that don't conform.

If we don't specify the rule at all, our local IDEs/profiles may each default to different values. This wouldn't matter in a lot of cases but if someone had it set to warning/error, that will cause issues once we start enforcing treat warnings as errors. 

There is a `silent` severity which will effectively ignore this rule. I did debate setting it to that but ultimately chose suggestion because I do actually think it increases readability.  We should probably get a list of all rules that have sparked discussion and have a quick meeting on them :)